src/realtime_calls.py

```
@@ -266,14 +321,98 @@ def _sanitize(self, call: dict) -> dict:
+            if isinstance(value, str):
+                return sanitize_command(self.redact_text(value))[:MAX_RAW_DATA_CHARS]
+
+            def fallback(item):
+                model_dump = getattr(item, "model_dump", None)
+                if callable(model_dump):
+                    return model_dump(mode="json")
+                return str(item)
+
+            text = json.dumps(value, ensure_ascii=False, indent=2, default=fallback)
+            return sanitize_command(self.redact_text(text))[:MAX_RAW_DATA_CHARS]
+        except Exception as exc:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)